src/tools/compressor/compressor.cpp

```diff
@@ -76,14 +80,19 @@ int main(int argc, char **argv)
         }
 
         if (output_file.Write(output_data.get(), output_size) != output_size) {
-            std::cerr << "Failed to write output data completly" << std::endl;
+            std::cerr << "Failed to write output data completely" << std::endl;
             return EXIT_FAILURE;
         }
     } else {
         CompressionType type = CompressionManager::Get_Prefered_Compression();
         if (result.count("type") > 0) {
-            auto compr_name = result["type"].as<std::string>();
-            type = CompressionManager::Get_Compression_Type(compr_name.c_str(), compr_name.size() + 5);
+            const auto compr_name = result["type"].as<std::string>();
+            if (compr_name.size() < 4) {
```

Comment:
Is this check really necessary? Wouldn't `CompressionManager::Get_Compression_Type` already cover this by returning `COMPRESSION_NONE` when size is smaller 4?

Reply:
We're not using this function as intended (it should be used on compressed data). However we use it on the `--type` parameter to just get the compression type

Reply:
How about we improve `Get_Compression_Type` to take a char* and do a strcmp instead?